components/leds.py

```diff
@@ -186,87 +164,68 @@ def setup(self) -> None:
         self.choose_morse_message()
         self.leds.setLength(self.led_length)
         self.single_led_data = wpilib.AddressableLED.LEDData()
-        self.single_led_data.setHSV(*self.color)
+        self.single_led_data.setHSV(*self.color[0])
         self.leds_data = [self.single_led_data] * self.led_length
         self.leds.setData(self.leds_data)
         self.leds.start()
 
-    def set_color(self, color: LedColors):
-        self.color = color.value
-
-    def set_piece(self, piece: PieceColour):
-        self.set_color(piece.value)
-
-    def set_state(self, state: RobotState):
-        if state == RobotState.PICKED_UP_PIECE:
-            self.pattern = DisplayType.SOLID
-        elif state == RobotState.LOOKING_FOR_PIECE:
-            self.pattern = DisplayType.HALF_HALF
-        elif state == RobotState.OTHER:
-            self.set_color(LedColors.OFF)
-
-    def set_intake_side(self, side: PickupFromSide):
-        self.side = side
+    def set_color(self, color: list[LedColors]):
+        colors = color * (self.led_length // len(color))
```

Comment:
Almost everywhere passes a list of length ~ `led_length`, but not all of them. Either this functionality should be consistently used, or not.